### DISCUSSION TITLE
*Ts syntax in type comments

### DISCUSSION THREAD

**Discussion by superbobry** [python/typing#1452] [Q&A] [answered]
Is it valid to use the `*Ts` syntax for unpacking a tuple type variable in type comments. For example, should type checkers treat

```
from typing import TypeVarTuple
T = TypeVarTuple("T")

def f(
    *args  # type: *Ts
):
    # (...) -> tuple[*Ts]
```

as if it was

```
def f(*args: *Ts) -> tuple[*Ts]: ...
```

?

**Comment by erictraut** [comment, marked as answer]
Type comments have been deprecated for a long time. They are still (partially) supported by some type checkers for old code that needs to remain compatible with Python 2, but you shouldn't be using them for new code. `TypeVarTuple` is a relatively new feature, so I wouldn't expect tools to support this with type comments.

If you want to use `TypeVarTuple` with a version of Python prior to 3.11, you can use `typing_extensions.Unpack`.

```python
from typing_extensions import Unpack, TypeVarTuple

Ts = TypeVarTuple("Ts")

def f(*args: Unpack[Ts]) -> Unpack[Ts]:
    ...
```

**Reply by superbobry** [reply]
Thanks, Eric! I think it is unlikely anyone will try to use type comments with `TypeVarTuple`s, but I thought I would ask anyway, since this is not mentioned at all in [PEP-646](https://peps.python.org/pep-0646/#unpack-for-backwards-compatibility).
